pkg/pipeline/pipeline.go

```diff
@@ -66,7 +68,12 @@ func newPipelineFromIngester(cfg *config.ConfigFileStruct, ing ingest.Ingester)
 	if err := builder.readStages(); err != nil {
 		return nil, err
 	}
-	return builder.build()
+	pipeline, err := builder.build()
+	if err != nil {
+		return nil, err
+	}
+	pipeline.configWatcher, err = newPipelineConfigWatcher(cfg, pipeline.pipelineEntryMap)
```

Comment:
Maybe I missed something somewhere, but I guess the config watcher shouldn't be initialized if it isn't configured? e.g. if FLP doesn't run in a kube environment ?

Reply:
newPipelineConfigWatcher now check if it needs to be created and return nil if it does not need it.